packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -327,4 +334,47 @@ App.propTypes = {
     config: PropTypes.object
 }
 
+/**
+ * a hook that parallelly and individually fetches category based on the given ids
+ * @param ids - list of categories ids to fetch
+ * @param queryOptions -  react query options
+ * @return list of react query results
+ */
+export const useCategoryBulk = (ids = [], queryOptions) => {
+    const api = useCommerceApi()
+    const {getTokenWhenReady} = useAccessToken()
+    const {
+        app: {commerceAPI}
+    } = getConfig()
+    const {
+        parameters: {organizationId}
+    } = commerceAPI
+    const {site} = useMultiSite()
+
+    const queries = ids.map((id) => {
+        return {
+            queryKey: queryKeyHelpers.getCategory.queryKey({
+                id,
+                levels: 2,
+                organizationId,
+                siteId: site.id
+            }),
+            queryFn: async () => {
+                const token = await getTokenWhenReady()
+                const res = await api.shopperProducts.getCategory({
+                    parameters: {id, levels: 2},
+                    headers: {
+                        authorization: `Bearer ${token}`
+                    }
+                })
+                return res
+            },
+            enabled: queryOptions.enabled && !!id,
+            ...queryOptions
```

Comment:
❗ This will _require_ users to do `useCategoryBulk(ids, {enabled: true})` in order to enable the query. I think this will also require `{enabled: false}`, otherwise the hook might throw an error. (I know `useQuery` throws if `enabled` is not a boolean, but I don't know about `useQueries`.)

To allow `queryOptions` to be an optional parameter, there are three options:
```suggestion
            enabled: Boolean(id),
            ...queryOptions
```
This is "enabled if `id` is set, unless the user overrides." The user can disable when there's an `id` or enable if it is missing.

```suggestion
            ...queryOptions,
            enabled: queryOptions.enabled !== false && Boolean(id)
```
This is "enabled if `id` is set and the user did not _disable_ the query." The user disable when there's an `id`, but cannot enable if it is missing.


```suggestion
            ...queryOptions,
            enabled: queryOptions.enabled !== true && Boolean(id)
```
This is "enabled when the user explicitly enables it _and_ there's an `id`." The user _must_ pass `{enabled: true}` to trigger the query, but it only works when there's also an `id`. (We probably don't want this option.)

Reply:
To me, the second one seems better. Unless anyone objects I'll commit that